streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1522,6 +1525,92 @@ private void maybeScheduleFollowupRebalance(final long encodedNextScheduledRebal
         }
     }
 
+    private AssignmentError validateTaskAssignment(final ApplicationState applicationState,
+                                                   final TaskAssignment taskAssignment) {
+        final Collection<KafkaStreamsAssignment> assignments = taskAssignment.assignment();
+
+        // Check for AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES
+        final Set<TaskId> activeTasks = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (task.type() != KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    continue;
+                }
+
+                if (activeTasks.contains(task.id())) {
+                    return AssignmentError.ACTIVE_TASK_ASSIGNED_MULTIPLE_TIMES;
+                }
+                activeTasks.add(task.id());
+            }
+        }
+
+        // Check for ACTIVE_AND_STANDBY_TASK_ASSIGNED_TO_SAME_KAFKASTREAMS
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            final Set<TaskId> activeTasksForAssignment = new HashSet<>();
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (task.type() == KafkaStreamsAssignment.AssignedTask.Type.ACTIVE) {
+                    activeTasksForAssignment.add(task.id());
+                }
+            }
+
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (task.type() == KafkaStreamsAssignment.AssignedTask.Type.STANDBY) {
+                    if (activeTasksForAssignment.contains(task.id())) {
+                        return AssignmentError.ACTIVE_AND_STANDBY_TASK_ASSIGNED_TO_SAME_KAFKASTREAMS;
+                    }
+                }
+            }
+        }
+
+        // Check for INVALID_STANDBY_TASK
+        final Set<TaskId> standbyTasksInOutput = new HashSet<>();
+        for (final KafkaStreamsAssignment assignment : assignments) {
+            for (final KafkaStreamsAssignment.AssignedTask task : assignment.assignment()) {
+                if (task.type() == KafkaStreamsAssignment.AssignedTask.Type.STANDBY) {
+                    standbyTasksInOutput.add(task.id());
+                }
+            }
+        }
+        for (final TaskInfo task : applicationState.allTasks()) {
+            if (!task.isStateful() && standbyTasksInOutput.contains(task.id())) {
+                return AssignmentError.INVALID_STANDBY_TASK;
+            }
+        }
+
+        // Check for MISSING_PROCESS_ID
```

Comment:
nit: no need for the comments, "the code should stand for itself" as one of the committers will always say. We typically prefer to limit inline comments to things that explain "why" some code is doing something, not "what " some code is doing